chia/wallet/wallet_node.py

```diff
@@ -667,10 +667,11 @@ def is_new_state_update(cs: CoinState) -> bool:
 
         # The number of coin id updates are usually going to be significantly less than ph updates, so we can
         # sync from 0 every time.
-        continue_while = True
-        all_coin_ids: List[bytes32] = await self.get_coin_ids_to_subscribe(0)
         already_checked_coin_ids: Set[bytes32] = set()
-        while continue_while:
+        while not self._shut_down:
+            all_coin_ids = await self.get_coin_ids_to_subscribe(0)
```

Comment:
It might contain new ones if we loop again from my understanding? The description of the method says:

```
"""
Sync algorithm:
    - Download and verify weight proof (if not trusted)
    - Roll back anything after the fork point (if rollback=True)
    - Subscribe to all puzzle_hashes over and over until there are no more updates
    - Subscribe to all coin_ids over and over until there are no more updates
    - rollback=False means that we are just double-checking with this peer to make sure we don't have any missing transactions, so we don't need to rollback
"""
```